### PR TITLE
[ResourceItem] Fix checkbox tap target

### PR DESCRIPTION
**Pull request by gwyneplaine**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/web/issues/92746 for the `ResourceItem` component.

### WHAT is this pull request doing?

* Previously the checkbox had a small clickable / tappable area that didn't cover the entirety of its visual space, this led to UX bugs where tapping on the checkbox would appear to trigger events on the larger container. This PR simplifies the ResourceItem component markup and enlarges the tap target to what it should be based on visual space.

Before (without media):
Screenshot 2023-06-29 at 2 23 29 pm

After (without media): 
Screenshot 2023-06-29 at 2 23 34 pm

Before (with media): 
Screenshot 2023-06-29 at 2 23 50 pm

After (with media) 
Screenshot 2023-06-29 at 2 23 24 pm

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/ResourceItem/ResourceItem.tsx [hunk 1]
```diff
@@ -162,42 +161,57 @@ class BaseResourceItem extends Component<CombinedProps, State> {
...
-                  paddingInlineEnd="3"
-                  paddingBlockStart={polarisSummerEditions2023 ? '2' : '3'}
-                  paddingBlockEnd="2"
+          {({polarisSummerEditions2023}) => {
+            if (polarisSummerEditions2023) {
+              return (
+                <div
+                  className={styles.CheckboxBleed}
+                  onChange={this.handleLargerSelectionArea}
+                  onClick={this.handleLargerSelectionArea}
```
Comment (jesstelford) [inline polaris-react/src/components/ResourceItem/ResourceItem.tsx:187]: Should this match the non-se23 logic by using `{preventDefault}`?

File: polaris-react/src/components/ResourceItem/ResourceItem.scss [hunk 2]
```diff
@@ -11,6 +11,41 @@
...
+// This class is only applied when the polarisSummerEditions2023 flag is enabled,
+// as such it doesn't need the #{$se23} selector.
+.ContainerWrapper {
+  // This class replaces the <Box/> component usage in ResourceItems 'containerMarkup' elements
+  // We want to specify a custom css property to ensure that the .CheckboxBleed padding and negative margin
+  // are calculated based on the padding on the parent container element.
+  // Box does not let us specify custom css properties unfortunately.
+
+  // stylelint-disable -- Polaris component custom properties
+  --pc-item-padding: var(--p-space-3);
```
Comment (jesstelford) [inline polaris-react/src/components/ResourceItem/ResourceItem.scss]: Should this be `--pc-resource-item-padding`?

File: polaris-react/src/components/ResourceItem/ResourceItem.stories.tsx [hunk 3]
```diff
@@ -25,7 +25,8 @@ export function Default() {
         items={[
           {
             id: '6',
-            url: '#',
+            // url: '#',
+            onClick: () => console.log('BOOP'),
```
Comment (jesstelford) [inline polaris-react/src/components/ResourceItem/ResourceItem.stories.tsx]: Can revert.

**Comment by mattkubej** [comment]
Is there a spin instance for this? I'm wondering what the hover state behavior is when hovering on the larger touch area, but not over the checkbox. From looking at the code, it seems we're likely capturing the clicks, but not applying the hover state. However, should it and should it have a hover hand?

**Comment by jesstelford** [comment]
No spinstance, but there's the Storybook that's published by CI from the PR: https://5d559397bae39100201eedc1-sienzkmtmv.chromatic.com/?path=/story/all-components-resourceitem--default&globals=polarisSummerEditions2023:true

The hover hand is present when hovering over the enlarged checkbox area, but you're correct that the checkbox isn't getting the hover state applied.

**Comment by jesstelford** [comment]
I can think of two possible ways to solve the hover state:

1. Allow the hover state to be controlled on `<Checkbox hovered={<boolean>}>`
    - This way ResourceItem can detect the hover with mouse events and toggle hover on/off manually
    - [Here's an implementation](https://github.com/Shopify/polaris/pull/9576/files?w=1) which does it
        - Look for "Storybook Publish" in the CI checks section to get a link to a Storybook, then look at the ResouceItem stories
1. Move this "bleed" logic up into the `<Checkbox>` component
    - That way it can handle it natively without having to expose controlled `hovered` props.
    - Solves it for all future uses of Checkbox which might need a larger tap target too.
    - [Here's an implementation](https://github.com/Shopify/polaris/pull/9593) which does it

**Comment by mattkubej** [comment]
I'm inclined to think this should be a native behavior or at least configured/enabled via props of the Checkbox component. Within a similar use case, I've been able to accommodate this for the time being by wrapping the Checkbox component in a div with a class, then adding a selector on the class to target the label of Checkbox and set fixed dimensions. Hovering the label results in triggering the hover state of the Checkbox. This does feel rather hacky and I think it would be more desirable to have a consistent touch target across uses of this component.

**Comment by jesstelford** [comment]
@mattkubej @gwyneplaine Here's an implementation adding a `<Checkbox bleed>` prop which meets the criteria of growing the clickable area to fill the space available while maintaining correct `:hover` states: https://github.com/Shopify/polaris/pull/9593

**Comment by jesstelford** [comment]
Superseded by https://github.com/Shopify/polaris/pull/9593

**Comment by jesstelford** [comment]
Superseded by https://github.com/Shopify/polaris/pull/9593

**Pull request closed without merging**